File: history/0.1.2/tatala-java/src/com/qileyuan/tatala/socket/to/helper/ByteMapHelper.java
package com.qileyuan.tatala.socket.to.helper;

import java.util.Iterator;
import java.util.Map;

import com.qileyuan.tatala.socket.io.TransferOutputStream;
import com.qileyuan.tatala.socket.to.TransferObject;
import com.qileyuan.tatala.socket.util.TransferUtil;

public class ByteMapHelper {
	public static int getByteArrayLength(Map<String, Byte> byteMap){
		int blength = 0;
		Iterator<String> keyIt = byteMap.keySet().iterator();
		while (keyIt.hasNext()) {
			// type
			blength += TransferUtil.getLengthOfByte();
			// key
			String key = (String) keyIt.next();
			blength += TransferUtil.getLengthOfString(key);
			// value
			blength += TransferUtil.getLengthOfByte();
		}
		return blength;
	}
	
	public static void getByteArray(Map<String, Byte> byteMap, TransferOutputStream touts) {
		Iterator<String> keyIt = byteMap.keySet().iterator();
		while (keyIt.hasNext()) {
			// type
			touts.writeByte(TransferObject.DATATYPE_BYTE);
			// key
			String key = (String) keyIt.next();
			touts.writeString(key);
			// value
			byte value = byteMap.get(key);
			touts.writeByte(value);
		}
	}
}

File: history/0.2.0/tatala-client-csharp/src/QiLeYuan/Tatala/socket/to/helper/ByteMapHelper.cs
using System;
using System.Collections.Generic;
using QiLeYuan.Tatala.socket.io;
using QiLeYuan.Tatala.socket.util;

namespace QiLeYuan.Tatala.socket.to.helper {

    public class ByteMapHelper {
        public static int getByteArrayLength(Dictionary<String, Byte> byteMap) {
            int blength = 0;
            foreach (KeyValuePair<String, Byte> item in byteMap) {
                // type
                blength += TransferUtil.getLengthOfByte();
                // key
                String key = item.Key;
                blength += TransferUtil.getLengthOfString(key);
                // value
                blength += TransferUtil.getLengthOfByte();
            }
            return blength;
        }

        public static void getByteArray(Dictionary<String, Byte> byteMap, TransferOutputStream touts) {
            foreach (KeyValuePair<String, Byte> item in byteMap) {
                // type
                touts.writeByte(TransferObject.DATATYPE_BYTE);
                // key
                String key = item.Key;
                touts.writeString(key);
                // value
                byte value = item.Value;
                touts.writeByte(value);
            }
        }
    }
}